File: 108-Convert-Sorted-Array-to-Binary-Search-Tree/solution.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
typedef struct TreeNode TreeNode; 

TreeNode* convert(int *nums, int start, int end)
{
    if (start>end){
        return NULL;
    }
    int mid = (start+end)/2;
    TreeNode *node = malloc(sizeof(TreeNode));
    node->val = nums[mid];
    node->left = convert(nums, start, mid-1);
    node->right = convert(nums, mid+1, end);
    return node;
}

struct TreeNode* sortedArrayToBST(int* nums, int numsSize) {
    return convert(nums, 0, numsSize-1);
}

File: 108-Convert-Sorted-Array-to-Binary-Search-Tree/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        return self.sort(nums, 0, len(nums)-1)
    
    def sort(self, nums, start, end):
        if start > end:
            return None
        mid = (start + end) / 2
        node = TreeNode(nums[mid])
        if start == end:
            return node
        node.left = self.sort(nums, start, mid-1)
        node.right = self.sort(nums, mid+1, end)
        return node